packages/admin/blocks-admin/src/blocks/VimeoVideoBlock.tsx

```diff
@@ -0,0 +1,109 @@
+import { Field, FieldContainer, FinalFormInput, FinalFormRadio, FinalFormSwitch } from "@comet/admin";
+import { Box, FormControlLabel } from "@mui/material";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { VimeoVideoBlockData, VimeoVideoBlockInput } from "../blocks.generated";
+import { BlocksFinalForm } from "../form/BlocksFinalForm";
+import { SelectPreviewComponent } from "../iframebridge/SelectPreviewComponent";
+import { useAdminComponentPaper } from "./common/AdminComponentPaper";
+import { createBlockSkeleton } from "./helpers/createBlockSkeleton";
+import { BlockCategory, BlockInterface } from "./types";
+
+const isValidVimeoIdentifier = (value: string) => {
+    const urlRegEx = /^(https?:\/\/)?((www\.|player\.)?vimeo\.com\/?(showcase\/)*([0-9a-z]*\/)*([0-9]{6,11})[?]?.*)$/;
+    const idRegEx = /^([0-9]{6,11})$/;
+
+    const urlMatch = urlRegEx.test(value);
+    const idMatch = idRegEx.test(value);
+
+    return urlMatch || idMatch;
+};
+
+const validateIdentifier = (value?: string) => {
+    if (!value) return undefined;
+
+    return value && isValidVimeoIdentifier(value) ? undefined : (
+        <FormattedMessage id="comet.blocks.vimeoVideo.validation" defaultMessage="Should be a valid Vimeo URL or identifier" />
+    );
+};
+
+export const VimeoVideoBlock: BlockInterface<VimeoVideoBlockData, VimeoVideoBlockData, VimeoVideoBlockInput> = {
+    ...createBlockSkeleton(),
+
+    name: "VimeoVideo",
+
+    displayName: <FormattedMessage id="blocks.vimeoVideo" defaultMessage="Video (Vimeo)" />,
+
+    defaultValues: () => ({ vimeoIdentifier: "", autoplay: false, showControls: true, loop: false, aspectRatio: "16X9" }),
+
+    category: BlockCategory.Media,
+
+    // !vimeoIdentifier to allow saving empty string
+    isValid: ({ vimeoIdentifier }) => true,
```

Comment:
Please re-add the validation here if vimeoIdentifiert isn't empty

Reply:
https://github.com/vivid-planet/comet/pull/2068/commits/d651e96822f249bec408d15559945496be6d5b31